HHS/Head-Start-TTADP
# [TTAHUB-2335] Cache similarity API results in DB table

## Description of change

Simple cache of similarity results in ready-to-access format. 
- Built on demand (no mass similarity job needed, since the API returns results very quickly) upon visits to the RTR
- Invalidated on goal name change, creation, or deletion
- Stores the user's judgment upon the goal group (not a match, partial match, full match)

## How to test

1. View RTR, with similarity results listed on the goals table (listed under "We found groups of similar goals that might be duplicates. To view and manage these goals, select a goal group")
2. Click one of those links
3. Select "none are duplicates"
4. Return to the RTR, note that goal group no longer appears in the results

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-2335


## Checklists

### Every PR



### Production Deploy


### After merge/deploy
